.github/workflows/nightly-build.yml

```
@@ -0,0 +1,116 @@
+#===============================================================================
+# Copyright contributors to the oneDAL project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#===============================================================================
+
+name: Nightly-build
+
+on:
+  schedule:
+    - cron: '0 21 * * *'
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  build_lnx:
+    name: build Linux
+    if: github.repository == 'oneapi-src/oneDAL'
+    runs-on: ubuntu-latest
```

Accidental failures due to non-pinned image version might cause interruption of development process.
```suggestion
    runs-on: ubuntu-22.04
```
```suggestion
    runs-on: ubuntu-24.04
```